### PR TITLE
added soundex function - closes zio#660

### PR DESCRIPTION
**Pull request by lukemcphee**

### CODE REVIEW

File: mysql/src/test/scala/zio/sql/mysql/FunctionDefSpec.scala [hunk 1]
```diff
@@ -121,6 +121,47 @@ object FunctionDefSpec extends MysqlRunnableSpec with ShopSchema {
...
+      val expected = "R163"
+      
+      
+      val testResult = execute(query)
+
+      val assertion = for {
+        r <- testResult.runCollect
+      } yield assert(r.head)(equalTo(expected))
+
+      assertion.mapErrorCause(cause => Cause.stackless(cause.untraced))
```
Comment (sviezypan) [inline mysql/src/test/scala/zio/sql/mysql/FunctionDefSpec.scala:163]: let's change to
assertZIO(testResult.runHead.some)(equalTo(expected))

File: mysql/src/test/scala/zio/sql/mysql/FunctionDefSpec.scala [hunk 2]
```diff
@@ -121,6 +121,47 @@ object FunctionDefSpec extends MysqlRunnableSpec with ShopSchema {
...
+
+      val resultForRobert = execute(queryForRobert)
+      val resultForRupert = execute(queryForRupert)
+
+      val assertion = for {
+        robertResult <- resultForRobert.runCollect
+        rupertResult <- resultForRupert.runCollect
+      } yield assert(robertResult.head.equals(rupertResult.head))(equalTo(true))
+
+      assertion.mapErrorCause(cause => Cause.stackless(cause.untraced))
```
Comment (sviezypan) [inline mysql/src/test/scala/zio/sql/mysql/FunctionDefSpec.scala:150]: Lets simplify
```
val assertion = for {
        robertResult <- resultForRobert.runCollect
        rupertResult <- resultForRupert.runCollect
      } yield assert(robertResult.head.equals(rupertResult.head))(equalTo(true))

      assertion.mapErrorCause(cause => Cause.stackless(cause.untraced))
```

to just
```
for {
        robertResult <- resultForRobert.runCollect
        rupertResult <- resultForRupert.runCollect
      } yield assert(robertResult.head.equals(rupertResult.head))(equalTo(true))
```

File: mysql/src/test/scala/zio/sql/mysql/FunctionDefSpec.scala [hunk 3]
```diff
@@ -121,6 +121,47 @@ object FunctionDefSpec extends MysqlRunnableSpec with ShopSchema {
...
+
+      val resultForRobert = execute(queryForRobert)
+      val resultForTam    = execute(queryForTam)
+
+      val assertion = for {
+        robertResult <- resultForRobert.runCollect
+        tamResult    <- resultForTam.runCollect
+      } yield assert(robertResult.head.equals(tamResult.head))(equalTo(false))
+
+      assertion.mapErrorCause(cause => Cause.stackless(cause.untraced))
```
Comment (sviezypan) [inline mysql/src/test/scala/zio/sql/mysql/FunctionDefSpec.scala:136]: same as below

**Review by sviezypan — CHANGES_REQUESTED** [review]
hi @lukemcphee I added some minor comments. Can you also please check why tests and formatting are failing?

**Review by jczuchnowski — APPROVED** [review]

**Pull request merged**